src/main.rs

```diff
@@ -753,13 +743,30 @@ async fn get_keypair(key: Option<PathBuf>) -> Result<Keypair> {
     }
 }
 
-async fn get_interactive(
-    hash: Hash,
-    opts: get::Options,
-    token: AuthToken,
-    out: Option<PathBuf>,
-) -> Result<()> {
-    progress!("Fetching: {}", Blake3Cid::new(hash));
+#[derive(Debug)]
+enum GetInteractive {
+    Ticket {
+        ticket: Ticket,
+        keylog: bool,
+    },
+    Hash {
+        hash: Hash,
+        opts: get::Options,
+        token: AuthToken,
+    },
+}
```

Comment:
This was the simplest way to add this functionality, it's not particularly elegant or well-named though.  But it works fine.  If you are disgusted however and have better ideas I'm happy to try different things.

Reply:
seems fie